src/asynqp/protocol.py

```diff
@@ -58,11 +60,17 @@ def connection_lost(self, exc):
 
     def heartbeat_timeout(self):
         """ Called by heartbeat_monitor on timeout """
+        assert not self._closed, "Did we not stop heartbeat_monitor on close?"
         log.error("Heartbeat time out")
         poison_exc = ConnectionLostError('Heartbeat timed out')
         poison_frame = frames.PoisonPillFrame(poison_exc)
         self.dispatcher.dispatch_all(poison_frame)
         # Spec says to just close socket without ConnectionClose handshake.
+        self.close()
+
+    def close(self):
+        assert not self._closed, "Why do we close it 2-ce?"
```

Comment:
Is this _possible_? If a user could provoke this condition by (ab)using the API, we should explicitly throw an exception.